server/test/auth/CiviFormProfileMergerTest.java

```diff
@@ -0,0 +1,134 @@
+package auth;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+
+import java.util.Collections;
+import java.util.Optional;
+import models.Account;
+import models.Applicant;
+import org.junit.Before;
+import org.junit.Test;
+import org.pac4j.core.profile.CommonProfile;
+import org.pac4j.core.profile.UserProfile;
+import org.pac4j.oidc.profile.OidcProfile;
+import repository.ResetPostgres;
+import repository.UserRepository;
+
+public class CiviFormProfileMergerTest extends ResetPostgres {
+
+  private static final String EMAIL_ATTR = "user_emailid";
+  private static final String EMAIL1 = "foo@bar.com";
+  private static final String EMAIL2 = "bar@foo.com";
+  private static final Long ACCOUNT_ID = 12345L;
+  private static final UserProfile USER_PROFILE = new CommonProfile();
+
+  private UserRepository repository;
+  private ProfileFactory profileFactory;
+
+  private OidcProfile oidcProfile;
+  private CiviFormProfileData civiFormProfileData;
+  private Applicant applicant;
+  private Account account;
+
+  private CiviFormProfileMerger civiFormProfileMerger;
+
+  @Before
+  public void setup() {
+    profileFactory = spy(instanceOf(ProfileFactory.class));
```

Comment:
We have full access to the database in these ResetPostgres tests.  Did you otherwise need spies rather than using the database directly?

In the other test you'll see things like 

profileFactory = instanceOf(ProfileFactory.class);
UserRepository repository = instanceOf(UserRepository.class);

Reply:
You're right, I started out trying to use the database, but then switched to using mocks because the database logic was already being tested in other unit test classes. I'll switched over to using Mockito fully.